### ISSUE TITLE
Migrate World server components to PSF-WorldServer

### ISSUE THREAD

**Issue by pschord** [psforever/PSF-LoginServer#13]
This project is supposed to be the login server only, but there are world server components.

Fix this to avoid confusion.

**Comment by pschord** [comment]
I've been doing more thinking on this issue:

1. Move all world server components to a separate project (using SBT) in the same Git repo as the current login server
 - **Advantages**
    + Would allow common code to be easily shared between the Login and World server (i.e. packets, Actors, logging, etc.).
    + All commits related to the "server" would be in one place
    + Running both servers for development would be easy
    + Unified issues and pull requests for the project
 - **Disadvantages**
    + Login server concerns could be buried in the world server
    + The source repository will be mixed in with game logic and common concerns
2. Separate all code for login and world servers
  - **Advantages**
    + Git commit logs would be separate and once the login server is done, it wont be touched
    + Game logic will be completely separate from boilerplate and packet code
    + Compiling will be faster as only common files will be present
    + Writing tests will become crucial to make sure library works
  - **Disadvantages**
    + All common files will be have to be in a separate git repository (this means 3 git repos) in order to share between
    + The login, world, and common files will have separate versions
    + Getting a client to connect to the server will be harder as two servers will have to be running
    + Contributing to the server would be much harder as a person now has to make commits to specific repos

**Comment by pschord** [comment]
```
[12:02 AM] DeathKill: I like option 2 on there
[12:03 AM] GeekOfWires: Option 2 has my vote
[12:06 AM] MooK: Option 2 + you can use package dependency management to handle the first and second disadvantages.
[12:07 AM] MooK: "shared-libs": "0.3.1"

"shared-libs": "0.3.0"
[12:07 AM] OzTurtle: But really, Option 2 I think works best. Getting the login server mostly complete doesn't seem to be the biggest hurdle. Thereafter most of the disadvantages are gone.
[12:07 AM] MooK: actually option 2 has no disadvantages if you do that, because it makes contribution easier
```

**Comment by Flosus** [comment]
I would also vote for splitting the repository. I think you can get lost in the code pretty easily, when you have one big repository. A folder structure isn't helping much, specially when using common code.

**Comment by pschord** [comment]
Looks like option 2 has won by popular vote.

Now to talk about how this would actually work.

* Repos
  - PSF-LoginServer - used solely for the login server concerns and tests
  - PSF-WorldServer - big and beefy world server code and tests
  - PSF-Common - common planetside types and packet code

Then everything can use PSF-Common as a versioned or SNAPSHOT dependency. This includes future projects such as bots or testing tools.

What needs to be done 

1. The common code needs to be set up in its own repository with it's own SBT build
2. We need to separate the world server code from the login server code

We also want continuous integration for all three repos, which means psforever/PSCrypto#2 needs to be fixed to avoid any headaches.

**Comment by pschord** [comment]
For the moment, this solution is scrapped as there aren't enough devs or churn to warrant the change.
A simple separation of *folders* and *subprojects* under one roof would suffice for now.

@Fate-JH Don't you have something like this already in the works?

**Comment by Fate-JH** [comment]
I had something stable but simple over in a branch of my `PSF-Server` repo, but the folder structure wasn't very well named.  I just moved around files and adjusted how the two sides (login and game) of the server started up so that `sbt pslogin/run` still worked exactly as normal even as either could be started individually.

I still hadn't implemented an "announce" channel so that the game server would send information to the login server and thus the server select functionality.  That's still hard-coded.

https://github.com/Fate-JH/PSF-Server/tree/ref-actor

**Comment by pschord** [comment]
Okay, I might go ahead and do that quickly in another PR.

Also the world and login server don't need to communicate except through a database.

**Comment by Mazo** [comment]
I think we can close this off for now with #531 being merged
